### PR TITLE
fix(code-examples): satisfy ESLint rules for datetime code examples

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2427/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2427/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2427/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2427/playground/)

File: apps/code-examples/src/app/code-examples/datetime/date-range-picker/custom-calculator/demo.component.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class DemoComponent {
       shortDescription: 'Date before today',
       type: SkyDateRangeCalculatorType.Before,
       validate: (value): ValidationErrors | null => {
-        if (value && value.endDate && value.endDate > new Date()) {
+        if (value?.endDate && value.endDate > new Date()) {
```
Comment (Blackbaud-ErikaMcVey) [inline apps/code-examples/src/app/code-examples/datetime/date-range-picker/custom-calculator/demo.component.ts:58]: can this just be `value?.endDate > new Date()`? undefined won't be greater than a date, if the property doesn't exist

Comment (Blackbaud-SteveBrush) [reply apps/code-examples/src/app/code-examples/datetime/date-range-picker/custom-calculator/demo.component.ts:58]: It cannot (with this ruleset). The ruleset requires that values being compared as numbers are actually number values.

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2434

**Pull request merged**